src/Test.UnitTests.Sarif/Visitors/SortingVisitorTests.cs

```diff
@@ -0,0 +1,306 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+using Microsoft.CodeAnalysis.Test.Utilities.Sarif;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Visitors
+{
+    public class SortingVisitorTests
+    {
+        private readonly Random random;
+
+        public SortingVisitorTests(ITestOutputHelper outputHelper)
+        {
+            random = RandomSarifLogGenerator.GenerateRandomAndLog(outputHelper);
+        }
+
+        [Fact]
+        public void SortingVisitor_ShuffleTest()
+        {
+            bool areEqual;
+            // create a test sarif log
+            SarifLog originalLog = CreateTestSarifLog(this.random);
+
+            SarifLog shuffledLog1 = originalLog.DeepClone();
+            SarifLog shuffledLog2 = originalLog.DeepClone();
+
+            // original log and cloned log should be same
+            areEqual = SarifLogEqualityComparer.Instance.Equals(originalLog, shuffledLog1);
+            areEqual.Should().BeTrue();
+
+            areEqual = SarifLogEqualityComparer.Instance.Equals(originalLog, shuffledLog2);
+            areEqual.Should().BeTrue();
+
+            areEqual = SarifLogEqualityComparer.Instance.Equals(shuffledLog1, shuffledLog2);
+            areEqual.Should().BeTrue();
+
+            // shuffle sarif log
+            ShuffleSarifLog(shuffledLog1, this.random);
```

Comment:
>ShuffleSarifLog [](http://example.com/codeflow?start=12&length=15)

I suggest changing this, accept a SARIF log, return a shuffled log. In your helper, deep clone the input log, and don't return until your helper verifies that the shuffled output differs from the input.

do 
{
 do work shuffling shuffledLog;
} while (shuffledLog == inputLog)

Reply:
updated the function.